### PR TITLE
Fixes nukes

### PR DESCRIPTION
**Pull request by Kyep**
Currently, if a nuke detonates at centcom, in an awaymission, etc, it won't harm anyone. Even if you're standing right next to it. Obviously, this is silly. This PR fixes that.

🆑 Kyep
fix: Nuclear fission explosives now kill all mobs on whatever zlevel they detonate, even if they detonate at centcom, in awaymissions, etc.
/🆑

### CODE REVIEW

**Comment by taukausanake** [comment]
Will this do anything to end the round, and does the nuke need to be anchored to a station turf in order to blow and end the round? I'm sure there are more questions that need to be asked on this but I can't think of them right now

**Comment by Lady-Luck** [comment]
This this fix the nuke not exploding on the cutscene?

**Comment by Kyep** [comment]
This fixes the 'people get cutscene, but nuke doesn't do damage when it explodes off station' thing.

**Comment by Lady-Luck** [comment]
When the nuke explodes off z level, the cutscene begins to play but it never does the explosion part of the cutscene. I was wondering if that was fixed too.

**Comment by Kyep** [comment]
When the nuke explodes off-station, currently you get a cutscene, and it shows the station NOT exploding, and nobody dies no matter what.

This PR fixes it so when the nuke explodes off station, you STILL get a cutscene, and it STILL shows the station not exploding... but everyone who IS around the nuke (wherever it is) dies.

**Comment by taukausanake** [comment]
Does it show an explosion way off away? That would be awesome

File: code/game/gamemodes/gameticker.dm [hunk 1]
```diff
@@ -251,24 +251,18 @@ var/round_start_time = 0
...
-					if(ghost && ghost.client) //Play the victims an uninterrupted cinematic.
-						ghost.client.screen += cinematic
-					CHECK_TICK
-			if(M && M.client) //Play the survivors a cinematic.
-				M.client.screen += cinematic
+	for(var/mob/M in mob_list)
+		M.buckled = temp_buckle
+		if(M.stat != DEAD)
+			var/turf/T = get_turf(M)
+			if(T && T.z == zlev && !istype(M.loc, /obj/structure/closet/secure_closet/freezer))
```
Comment (Fox-McCloud) [inline code/game/gamemodes/gameticker.dm]: use the proper atom check proc for z-levels please.

**Comment by Fox-McCloud** [comment]
Pretty sure this is going to break things, like the fake nuke.

It also means the cinematic verb is never going to kill anyone; a change in behavior over current.

File: code/game/gamemodes/gameticker.dm (hunk 1 above)
Comment (Kyep) [inline code/game/gamemodes/gameticker.dm]: And the 'proper atom check proc' is...?

**Comment by Kyep** [comment]
https://github.com/ParadiseSS13/Paradise/blob/2944debd3f564280e55b82f776acde8e2b612cbc/code/game/objects/items/toys.dm#L785 does not call station_explosion_cinematic.

Cinematic has been changed to still be deadly.

**Comment by FalseIncarnate** [comment]
I think this is a terrible idea, and will go into detail as to why:

To preface the following points, the assumption is that the nuke exploding is part of the nuclear operative game mode. Nuke explosions in other modes will be ignored for now as they are more of admin-intervention in the majority of cases, or cases where the nuke is expected to remain on station until it explodes.

If the nuke is moved sufficiently into space, the explosion does not kill the crew, as the nuke is considered "out of range". The exact conditions for this to be considered as true are calculated [here](https://github.com/ParadiseSS13/Paradise/blob/master/code/game/gamemodes/nuclear/nuclearbomb.dm#L365), though to save people the trouble of deciphering a long multi-conditional, the bomb must be within 40-ish tiles of any map edge on the station level to be considered a near-miss. In this event, the Crew is awarded a minor victory: the operatives blew something up, but not the station; so they all live to get blown up another day or suffer one of the many other gruesome fates that await our brave spessmen and spesswomen.

Now, as [this comment](https://github.com/ParadiseSS13/Paradise/blob/master/code/game/gamemodes/nuclear/nuclearbomb.dm#L388) suggests, the intent is that near-station detonations are meant to irradiate the entire station (we'll say z-level for simplicity), since the explosion itself can't reach the station but radiation travels quite fantastically through vacuum (if you doubt this, look at the Sun and/or stars next time you get a chance and you'll get a wonderful eyeful of it). The crew survives the detonation, but is STRONGLY encouraged to leave the station before the fallout reduces them to ~feral ghouls~ fat Swedes with Tourette's Syndrome and a lisp... and eventually kills those unfortunate enough to not be immune to radiation (Glory to Synthetica! Also the Diona can hang out too).

While I think that the nuke SHOULD kill anyone unfortunate enough to be caught in its blast radius (which is [80 tiles](https://github.com/ParadiseSS13/Paradise/blob/master/code/game/gamemodes/nuclear/nuclearbomb.dm#L345)), but not simply everyone within the same z-level as itself with the explosion. The rest of the z-level should then become heavily irradiated, which in itself will kill a lot of people while allowing the lucky few to escape (especially those who procure a radiation suit while waiting for the shuttle), thus holding truer to the obvious intent of the coder who placed the comment in the code I referenced in the previous point.

**Comment by Kyep** [comment]
@FalseIncarnate Restricting nukes to an 80-tile radius is a major change to how they work, and it would mean we had to replace the 'station blows up' cinematic. I don't see us doing that.

**Comment by FalseIncarnate** [comment]
I'm not saying restrict the nuke to an 80 tile radius, nor would doing so require replacing the cinematic.

What I'm saying is that if the nuke isn't on station, make it kill within 80 tiles and irradiate the entire level, rather than just kill everything on the level, so it won't kill the entire station on near-miss.

This would be actually quite trivial to accomplish: [this](https://github.com/ParadiseSS13/Paradise/blob/master/code/game/gamemodes/nuclear/nuclearbomb.dm#L376) handles the showing of the proper cinematic, which is what actually kills people (movies kill people!). [THIS](https://github.com/ParadiseSS13/Paradise/blob/67d5d69a6bb9d7cc095d00503359b3f32e194f9a/code/game/gamemodes/gameticker.dm#L254), however, is what handles missed explosions (aka, not killing everyone). Simply have it loop through everyone within `NUKERANGE` (defined in nuclearbomb.dm as 80, we can simply move this define or something if needed) of the bomb's location (which you'd want to obtain through either passing it to the proc as an argument or simply locating the nuke in the world) and killing them, then applying an irradiate effect to the entire z-level the bomb is on.

To give a rough example (via some psuedo-code because fuck copying that loop for an example):
```
if(off_station)
   var/obj/machinery/nuke/da_bomb
   for(var/obj/machinery/nuke/bomb in world)
      if(bomb.bomb_set && (bomb.timing == -1))
         da_bomb = bomb
         break
   if(da_bomb)
      for(var/mob/M in mob_list))   //ideally just the z-level, but we'll just use mob_list for this example
         //Irradiate them, turn them into glowing goo!
         if(M in range(NUKERANGE, da_bomb))
            //Kill them johnny, kill them all.
```

File: code/game/gamemodes/gameticker.dm (hunk 1 above)
Comment (Crazylemon64) [inline code/game/gamemodes/gameticker.dm]: `atoms_share_zlev(src, M)`

**Comment by Anticept** [comment]
@FalseIncarnate a nuclear blast near a space station would not significantly irradiate the inside of the station unless it breaches the hull (or they use really crappy materials, but then how does the atmosphere not blow the hull out?). The reason why this is a problem on the surface of earth, is the radioactive fallout which contains fissile materials. In space, these would not be a factor as most of it would just float away. The little bit that remains behind might settle on the hull, but is of little consequence as everyone is in radiation shielding suits when they go outside anyways.

The only places this might be an issue is fissile materials settling on windows exposed to space, but I bet they would be designed to filter out at least *some* of the high energy photons.

**Comment by taukausanake** [comment]
@Anticept What about the radiation fields the station keeps passing through? Would those have any similarities with nuclear radiation?

**Comment by Anticept** [comment]
I edited the post, for some reason I forgot the disclaimer pointing out real vs ss13 logic.

**Comment by TullyBurnalot** [comment]
Can't say I'm a fan of this PR.

As it stands, it essentially guarantees everyone gets killed, even if technically that means the Nukies lose if the nuke is not on-station, leading to really weird counter-intuitive results, since this doesn't seem to touch on victory conditions.

If it _did_ touch on victory conditions, it'd be a free pass for Nukies to get the disk and then plant it somewhere no one can reach it, making any last ditch effort at saving the station effectively pointless.

**Comment by tigercat2000** [comment]
As Tully said above, we don't believe the very rare edge cases are worth introducing changes that conflict with the round ending conditions.

**Pull request closed without merging**